src/lexer/scanner.go

```diff
@@ -267,36 +278,18 @@ func (s *Scanner) Scan() Token {
 			s.currentColumnFile = 1
 		}
 
-		if containsByte(s.charsToIgnore, currChar) {
-			s.dft.Next(currSymbol)
-			if s.dft.IsFinalState() {
-				tokenClass := s.getTokenClass(s.dft.GetCurrentState())
-				lexem := s.lexemBuffer
-				dataType := s.getDataType(string(lexem))
-
-				s.clearLexemBuffer()
-				s.dft.Reset()
-				s.file.Seek(-1, os.SEEK_CUR)
-
-				token := NewToken(tokenClass, string(lexem), dataType)
-				InsertSymbolTable(string(lexem), token)
-
-				return token
-			}
-		}
-
 		_, err = s.dft.Next(currSymbol)
 
 		if errors.Is(err, ErrorTransitionDoesNotExist) && s.dft.IsFinalState() {
 			tokenClass := s.getTokenClass(s.dft.GetCurrentState())
 			lexem := s.lexemBuffer
-			dataType := s.getDataType(string(lexem))
+			token := NewToken(tokenClass, string(lexem), NULL)
+			s.updateDataType(&token)
```

Comment:
Check if these two lines can be merged into one, considering that the update can be done inside the "NewToken" method.